src/general/GameWorld.cs

```diff
@@ -76,13 +76,21 @@ public GameWorld(WorldGenerationSettings settings, Species? startingSpecies = nu
             TimedEffects.RegisterEffect("volcanism", new VolcanismEffect(this));
             TimedEffects.RegisterEffect("nitrogen_control", new NitrogenControlEffect(this));
 
-            // Patch events. Their sequence SHOULD NOT be changed!
+            // Patch events. PatchEventsManager HAS to be the last one
             TimedEffects.RegisterEffect("global_glaciation_event",
                 new GlobalGlaciationEvent(this, random.Next64()));
             TimedEffects.RegisterEffect("meteor_impact_event",
                 new MeteorImpactEvent(this, random.Next64()));
             TimedEffects.RegisterEffect("underwater_vent_eruption",
-                new UnderwaterVentEruptionEffect(this, random.Next64()));
+                new UnderwaterVentEruptionEvent(this, random.Next64()));
+            TimedEffects.RegisterEffect("runoff_event",
+                new RunoffEvent(this, random.Next64()));
+            TimedEffects.RegisterEffect("upwelling_event",
+                new UpwellingEvent(this, random.Next64()));
+            TimedEffects.RegisterEffect("current_dilution_event",
+                new CurrentDilutionEvent(this, random.Next64()));
+            TimedEffects.RegisterEffect("patch_events_manager",
+                new PatchEventsManager(this, random.Next64()));
```

Comment:
To make old saves compatible again, I think this makes things slightly complicated. I think what should be done here is bumping up the serialization version and then on load checking if the version is < 2 and if it is then creating these new events and appending them to the data. That should make old saves work well with the new patch events manager. This is the power of the new saving system that should allow much easier save compatibility across versions.